### ISSUE TITLE
0.24.0 crashes on macOS, M1 Pro machine

### ISSUE THREAD

**Issue by yanzhang0219** [kovidgoyal/kitty#4420]
**Describe the bug**
Installed kitty via homebrew. Click the icon to open it but it crashes immediately. Tried to restart my laptop, the issue was still there. Also tried to download the macOS.dmg directly, and same issue happened. 

BTW, the previous version 0.23.1 worked well. 

**To Reproduce**
On macOS, run `brew install kitty`. Then click kitty icon.

**Environment details**
```
macOS version: 12.1
machine: MacBook Pro with Apple Silicon (M1 Pro)
```

**Additional context**
In iTerm2, run `kitty --config NONE`, same issue. 

Thanks.

**Comment by kovidgoyal** [comment]
I'm afraid I dont support homebrew, use the official kitty binaries.
https://sw.kovidgoyal.net/kitty/binary/

**Comment by yanzhang0219** [comment]
Okay. I will try that. Thank you very much. 
However, why the `macOS dmg` in the releases does not work neither? Thanks.

**Comment by kovidgoyal** [comment]
Doesnt crash for me. If it is actually crashing, post the crash report.

**Comment by yanzhang0219** [comment]
Tried the binaries already and it crashed again. How can I get the crash report? Please apologize for my limited knowledge.

**Comment by kovidgoyal** [comment]
When a program crashes on macOS, macOS will popup a dialog with a button
to see the crash report. If you arent getting that then it isnt
crashing.

**Comment by yanzhang0219** [comment]
Sorry for my inaccurate word. After clicking the icon or run `kitty --config NONE` from another terminal, the kitty window appears but quits immediately. No other messages or dialog popups. Please see the demo below. Thanks. 

https://user-images.githubusercontent.com/11582667/148013557-2b9b2968-6004-4b48-84b1-6ccadb51c8f8.mov

**Comment by kovidgoyal** [comment]
That will be because the shell is exiting. Run kitty as

kitty --hold

to see the error message from the shell.

**Comment by yanzhang0219** [comment]
Okay. After running `kitty --hold` in iTerm2, a kitty window is open, however, no error messages. 

<img width="1057" alt="image" src="https://user-images.githubusercontent.com/11582667/148014190-e4b591b2-472a-41a6-9cca-67ea01082e40.png">

**Comment by kovidgoyal** [comment]
I am totally confused. Are you saying running:

kitty

causes kitty to exit immediately but adding --hold causes it to work normally?

**Comment by page-down** [comment]
I noticed in the screenshot of the `kitty --hold` that the prompt is in red, does this mean that the shell rc is running with an error (non-zero exit code)?

Does this seem to be related to shell integration?

```shell
kitty --config=NONE -o shell_integration=disabled
```

**Comment by yanzhang0219** [comment]
Yes. In iTerm2, if I run `kitty` without any arguments given or `kitty --config NONE`, it exits immediately. If I run `kitty --hold`, it works well, i.e., a kitty window is open. 

I am confused as well because 0.23.1 version works perfectly, but 0.24.0 not. 

Thanks.

**Comment by kovidgoyal** [comment]
Add shell_integration disabled to your kitty.conf and you will be fine.

**Comment by yanzhang0219** [comment]
It works after adding `shell_integration disabled` in my kitty.conf. Does that mean the new feature, shell integration, cannot work on macOS?

**Comment by kovidgoyal** [comment]
No, it means you have something in your shell rc files that is breaking it.

**Comment by yanzhang0219** [comment]
I'm using zsh and how can I figure out what is breaking this cool feature quickly? Any ideas? Thank you so much.

**Comment by page-down** [comment]
You need to find what particular configuration in your shell rc that is causing the problem.

For example, comment out each line of the script and find the line that is causing the problem.

**Comment by yanzhang0219** [comment]
@page-down Okay. I will try to figure it out. I thought probably there would be some shortcuts I could go through to find it. :) Thanks.

**Comment by kovidgoyal** [comment]
There are no error messages from kitty. Your shell is failing, and its not printing any error messages, if it did, they would show up when using --hold. Why its doing that, I have no idea. In general I suggest you prune your zshrc to what you actually need/use/understand.

**Comment by yanzhang0219** [comment]
I figured it out. 

The [zimfw/input](https://github.com/zimfw/input) plugin makes kitty exit. I uninstalled it and kitty works well now. I have little knowledge about zsh so I have no idea which line in https://github.com/zimfw/input/blob/master/init.zsh causes this issue. I will open an issue at that repo to see whether its author @ericbn has ideas. 

Thank you guys for your patience and helping me solve this issue.

**Comment by yanzhang0219** [comment]
@kovidgoyal I find that line 113, https://github.com/zimfw/input/blob/master/init.zsh#L113, causes this issue. I have no ideas what is the meaning of this line. Do you think whether it can be fixed on kitty's side? Thanks a lot.

**Comment by kovidgoyal** [comment]
I highly doubt its that line causing the issue, indeed sourcing the entire init.zsh file causes no issues in kitty 0.24 for me with shell integration enabled.

**Comment by yanzhang0219** [comment]
I commented that line and then I can opened kitty and kitty worked well.

**Comment by kovidgoyal** [comment]
Yeah that's likely because that line is interactiong with something else in your zsh config. Use an empty zsh config with only init.zsh sourced and see if it still breaks.

**Comment by yanzhang0219** [comment]
After further investigation, I found that if I kept zimfw/input installed but uninstalled these two plugins [zsh-syntax-highlighting](https://github.com/zsh-users/zsh-syntax-highlighting) and [zsh-history-substring-search](https://github.com/zsh-users/zsh-history-substring-search). Kitty still works normally.

**Comment by kovidgoyal** [comment]
I have no idea. Using zsh-syntax-highlighting and zimfw/input works fine for me with kitty's zsh integration.

**Comment by yanzhang0219** [comment]
That's weird. Thanks anyway.

**Comment by yanzhang0219** [comment]
Just to mention that this issue was solved after upgrading to 0.24.1. 
Thanks @kovidgoyal
